catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java

```diff
@@ -899,12 +899,17 @@ private String findAnyMatchingNumericalField(String propertyName, String fieldSu
   }
 
   String getSortKey(String field) {
-    if (field.endsWith(SchemaFields.GEO_SUFFIX)) {
+    if (field.endsWith(SchemaFields.GEO_SUFFIX)
+        || (field.endsWith(SchemaFields.TEXT_SUFFIX) && caseInsensitiveSort())) {
       field = field + SchemaFields.SORT_SUFFIX;
     }
     return field;
   }
 
+  private boolean caseInsensitiveSort() {
+    return "true".equals(System.getProperty("solr.sort.case-insensitive"));
```

Comment:
❗ I realized the other day that similar code I had added to the Solr client could cause threading issue since System properties are backed by a hashtable which is synchronized.  In the other case the method would generally short circuit before reaching the system property check.  In this case, it will happen for every sorted geo and text field.

https://github.com/AdoptOpenJDK/openjdk-jdk8u/blob/master/jdk/src/share/classes/java/lang/System.java#L533

I would recommend loading this property into a field.  We can restart the bundle if we need to refresh the value.